spec/features/pictures_feature_spec.rb

```diff
@@ -0,0 +1,116 @@
+require 'rails_helper'
+
+feature 'Images' do
+  context 'no images have been uploaded' do
+    scenario 'should display a prompt to upload an image' do
+      visit '/photos'
+      expect(page).to have_content 'No images yet'
+      expect(page).to have_link 'Upload a new photo'
+    end
+  end
+
+  context 'images have been added' do
+    before do
+      Photo.create(title: 'Tomato')
+    end
+
+    scenario 'display images' do
+      visit '/photos'
+      expect(page).to have_content('Tomato')
+      expect(page).not_to have_content('No photos yet')
+    end
+  end
+
+  context 'viewing photos' do
+    let!(:tomato) { Photo.create(title:'Tomato') }
+    scenario 'titles are links' do
+      visit '/photos'
+      click_link 'Tomato'
+      expect(page).to have_content 'Tomato'
+      expect(current_path).to eq "/photos/#{tomato.id}"
+    end
+  end
+
+  context 'uploading images' do
+    before(:each) do
+      user = build :user
+      sign_up(user)
+    end
+
+    scenario 'user can upload image with title' do
+      visit '/photos'
+      click_link 'Upload a new photo'
+      fill_in 'Title', with: 'Tomato'
+      attach_file("photo[image]", "spec/asset_specs/photos/tomato.jpg")
+      click_button 'Upload'
+      expect(page).to have_selector 'img'
+      expect(page).to have_css ('img[src*="tomato.jpg"]')
+    end
+
+     xscenario 'image does not display if no image uploaded' do
+      visit '/photos'
+      click_link 'Upload a new photo'
+      fill_in 'Title', with: 'Tomato'
+      click_button 'Upload'
+      expect(page).not_to have_selector 'img'
+      expect(page).to have_content 'Please upload an image'
+    end
```

Comment:
`end` at 57, 4 is not aligned with `xscenario 'image does not display if no image uploaded' do` at 50, 5.